src/JustSaying/Messaging/Interrogation/IInterrogable.cs

```diff
@@ -0,0 +1,15 @@
+namespace JustSaying.Messaging.Interrogation
+{
+    /// <summary>
+    /// Provides unstructured interrogation. Implementations of this interface should return an anonymous object that
+    /// can be composed together into a root object.
+    /// </summary>
+    public interface IInterrogable
```

Comment:
I'm not sure I'm a fan of this interface to be honest. It's not useful programmatically other than a "dump all this stuff somewhere", as you can't do anything with the return type in code as you won't know what to expect.

Reply:
Let's discuss this afternoon 🙂